atom/common/native_mate_converters/net_converter.cc

```diff
@@ -165,6 +165,34 @@ v8::Local<v8::Value> Converter<net::HttpResponseHeaders*>::ToV8(
   return ConvertToV8(isolate, response_headers);
 }
 
+bool Converter<net::HttpResponseHeaders*>::FromV8(v8::Isolate* isolate,
+    v8::Local<v8::Value> val,
+    net::HttpResponseHeaders *out) {
+  if (!val->IsObject()) {
+    return false;
+  }
+  auto context = isolate->GetCurrentContext();
+  auto headers = v8::Local<v8::Object>::Cast(val);
+  auto keys = headers->GetOwnPropertyNames();
```

Comment:
Its better to add a helper in `mate::Dictionary` to retrieve the keys and use it here instead. Can simply this function.

Reply:
:+1:

Reply:
Can you show an example of what you mean? I've tried using `mate::Dictionary` but it doesn't seem to provide any methods to retrieve all keys so I end up having to use something like `headers.GetHandle()->GetOwnPropertyNames()` (which would result in similar code).

Reply:
Oh I meant adding a new method to `mate::Dictionary` that allows retrieving all the keys and work with it. I am fine with the existing converter but would be better if we can reuse stuff.

Reply:
Since this also involves sending a PR to another project, Can I do this in my next PR, after the necessary changes (EventSubscriberBase + Dictionary) are pulled into https://github.com/electron/native-mate?